clang/test/SemaSYCL/attr-add-ir-attributes-merge.cpp

```diff
@@ -4,43 +4,102 @@
 // attributes to generate are the same.
 
 void FunctionRedecl1();
+[[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, true)]] void FunctionRedecl1();  // expected-note {{conflicting attribute is here}}
 [[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, true)]] void FunctionRedecl1();
-[[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, true)]] void FunctionRedecl1();
-[[__sycl_detail__::add_ir_attributes_function("Attr2", "Attr1", true, 1)]] void FunctionRedecl1();  // expected-note {{conflicting attribute is here}}
-[[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, false)]] void FunctionRedecl1(); // expected-error {{attribute '__sycl_detail__::add_ir_attributes_function' is already applied with different arguments}} expected-note {{conflicting attribute is here}}
-[[__sycl_detail__::add_ir_attributes_function("Attr3", false)]] void FunctionRedecl1(){};           // expected-error {{attribute '__sycl_detail__::add_ir_attributes_function' is already applied with different arguments}}
+[[__sycl_detail__::add_ir_attributes_function("Attr2", "Attr1", true, 1)]] void FunctionRedecl1();
+[[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, false)]] void FunctionRedecl1(); // expected-error {{attribute '__sycl_detail__::add_ir_attributes_function' is already applied with different arguments}}
+[[__sycl_detail__::add_ir_attributes_function("Attr3", false)]] void FunctionRedecl1(){};
 
 [[__sycl_detail__::add_ir_attributes_function("Attr1", "Attr2", 1, true)]] void FunctionRedecl2();
 void FunctionRedecl2();
 
 void FunctionRedecl3();
-[[__sycl_detail__::add_ir_attributes_function({"Attr1", "Attr3"}, "Attr1", "Attr2", 1, true)]] void FunctionRedecl3();
+[[__sycl_detail__::add_ir_attributes_function({"Attr1", "Attr3"}, "Attr1", "Attr2", 1, true)]] void FunctionRedecl3();  // expected-note {{conflicting attribute is here}}
 [[__sycl_detail__::add_ir_attributes_function({"Attr1", "Attr3"}, "Attr1", "Attr2", 1, true)]] void FunctionRedecl3();
 [[__sycl_detail__::add_ir_attributes_function({"Attr1", "Attr3"}, "Attr2", "Attr1", true, 1)]] void FunctionRedecl3();
-[[__sycl_detail__::add_ir_attributes_function({"Attr3", "Attr1"}, "Attr1", "Attr2", 1, true)]] void FunctionRedecl3();  // expected-note {{conflicting attribute is here}}
+[[__sycl_detail__::add_ir_attributes_function({"Attr3", "Attr1"}, "Attr1", "Attr2", 1, true)]] void FunctionRedecl3();
 [[__sycl_detail__::add_ir_attributes_function({"Attr1", "Attr3"}, "Attr1", "Attr2", 1, false)]] void FunctionRedecl3(); // expected-error {{attribute '__sycl_detail__::add_ir_attributes_function' is already applied with different arguments}} expected-note {{conflicting attribute is here}}
```

Comment:
Sorry what is the conflict here? I might just be misunderstanding how filters work, but you said only the attributes in the filter list are applied, which means only Attr1 is applied right? Isn't it's value 1 in all cases?

Reply:
We resolve conflicts without applying the filter. The argument is that from clang's point-of-view, the values are still conflicting, even if they don't end up producing IR attributes. I don't think this behavior is necessarily strictly needed, but I like to think it fits the mental model.

Reply:
Ok. I defer to you since I am not very familiar with the usage/functionality of these attributes. Please add AST tests though and maybe a comment somewhere how filters work in this context. It is not intuitive.